libvast/vast/file.hpp

```diff
@@ -76,17 +76,14 @@ class file {
   /// Reads a given number of bytes in to a buffer.
   /// @param sink The destination of the read.
   /// @param size The number of bytes to read.
-  /// @param got The number of bytes read.
   /// @returns `caf::none` on success.
-  [[nodiscard]] caf::error read(void* sink, size_t size, size_t* got = nullptr);
+  [[nodiscard]] caf::error read(void* sink, size_t size);
```

Comment:
Is there a specific reason why we're discarding the number of bytes read here? We could just return `expected<size>` from this function and handle the error downstream, which is usually more appropriate.

Reply:
Yeah, this looks like an oversight to me.

Reply:
I followed your suggestion, it was reverted to return the number of bytes that were read.